internal/config/endpoints.go

```diff
@@ -25,176 +35,143 @@ type Endpoint struct {
-	},
-	"europe-west6": {
-		URL:        "https://europe-west6-5tkroniexa-oa.a.run.app",
-		Region:     "europe-west6",
-		RegionName: "Zurich",
-	},
-	"europe-west8": {
-		URL:        "https://europe-west8-5tkroniexa-oc.a.run.app",
-		Region:     "europe-west8",
-		RegionName: "Milan",
-	},
-	"europe-west9": {
-		URL:        "https://europe-west9-5tkroniexa-od.a.run.app",
-		Region:     "europe-west9",
-		RegionName: "Paris",
-	},
-	"europe-southwest1": {
-		URL:        "https://europe-southwest1-5tkroniexa-no.a.run.app",
-		Region:     "europe-southwest1",
-		RegionName: "Madrid",
-	},
-	"northamerica-northeast1": {
-		URL:        "https://northamerica-northeast1-5tkroniexa-nn.a.run.app",
-		Region:     "northamerica-northeast1",
-		RegionName: "Montréal",
-	},
-	"northamerica-northeast2": {
-		URL:        "https://northamerica-northeast2-5tkroniexa-pd.a.run.app",
-		Region:     "northamerica-northeast2",
-		RegionName: "Toronto",
-	},
-	"southamerica-east1": {
-		URL:        "https://southamerica-east1-5tkroniexa-rj.a.run.app",
-		Region:     "southamerica-east1",
-		RegionName: "São Paulo",
-	},
-	"southamerica-west1": {
-		URL:        "https://southamerica-west1-5tkroniexa-tl.a.run.app",
-		Region:     "southamerica-west1",
-		RegionName: "Santiago",
-	},
-	"us-central1": {
-		URL:        "https://us-central1-5tkroniexa-uc.a.run.app",
-		Region:     "us-central1",
-		RegionName: "Iowa",
-	},
-	"us-east1": {
-		URL:        "https://us-east1-5tkroniexa-ue.a.run.app",
-		Region:     "us-east1",
-		RegionName: "South Carolina",
-	},
-	"us-east4": {
-		URL:        "https://us-east4-5tkroniexa-uk.a.run.app",
-		Region:     "us-east4",
-		RegionName: "North Virginia",
-	},
-	"us-east5": {
-		URL:        "https://us-east5-5tkroniexa-ul.a.run.app",
-		Region:     "us-east5",
-		RegionName: "Columbus",
-	},
-	"us-west1": {
-		URL:        "https://us-west1-5tkroniexa-uw.a.run.app",
-		Region:     "us-west1",
-		RegionName: "Oregon",
-	},
-	"us-west2": {
-		URL:        "https://us-west2-5tkroniexa-wl.a.run.app",
-		Region:     "us-west2",
-		RegionName: "Los Angeles",
-	},
-	"us-west3": {
-		URL:        "https://us-west3-5tkroniexa-wm.a.run.app",
-		Region:     "us-west3",
-		RegionName: "Salt Lake City",
-	},
-	"us-west4": {
-		URL:        "https://us-west4-5tkroniexa-wn.a.run.app",
-		Region:     "us-west4",
-		RegionName: "Las Vegas",
-	},
+type innerEndpoint struct {
+	//Used to unmarshal properly formatted JSON obtained from /api/endpoint JSON
+	URL        string
+	Region     string
+	RegionName string
+}
+
+var allEndpoints map[string]Endpoint
+
+func getJson(url string, target interface{}) error {
+	r, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	return json.NewDecoder(r.Body).Decode(target)
+
+}
+
+func GetEndpoints() map[string]Endpoint {
+	//TODO detect when running in Cloud Run and GenerateConfigFromAPI()
+	if len(allEndpoints) == 0 {
+		allEndpoints = GenerateConfigFromEndpoints()
+	}
```

Comment:
I'm a little nervous about caching this in a variable without a specific way of refreshing it (or locking it behind a mutex for concurrency).

Why not drop this method alltogether and allow the callers to determine which function they should use as opposed to trying to detect if running in Cloud Run and populating them differently?

Reply:
Dropped the `GetEndpoints()` method in favor of letting the caller determine which function they should use.

I'll add caching ttl + mutex for the URL map, as there is a performance/latency cost with fetching the endpoint map, and the workers fetch it for each single ping (so a total of 10 pings x 34 regions = 340 calls in a session)

Reply:
Thought I'd clarify: I'll add caching ttl + mutex for the URL map for the `/api/endpoints` handler on the Cloud Run side only.

On the cli, I don't see issues with caching the value in memory without a refresh mechanism, since the cache ttl is the duration of the process anyway. Let me know if I'm missing something.

Reply:
Resolved with following changes:
- Dropped the GetEndpoints() method in favor of letting the caller determine which function they should use
- Added local caching + mutex using [go-cache](https://github.com/patrickmn/go-cache) for web /api/endpoints
- Added caching in local var for cli as value is WO/RM and TTL is limited to the execution of the cli (<1min)